repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,193 @@
+      • Synchronize customer (partner) across all repairs in the same group
+      • Cascade confirmation/cancellation actions within the group
+      • Create a single sale order for all repairs of one group
+    """
+
+    _inherit = "repair.order"
+
+    group_id = fields.Many2one(
+        "repair.order.group",
+        string="Repair Group",
+        ondelete="set null",
+        check_company=True,
+        index=True,
+        readonly=True,
+        help="Repairs sharing the same group can be processed and quoted together.",
+    )
+
+    grouped_repair_ids = fields.One2many(
+        "repair.order",
+        compute="_compute_grouped_repair_ids",
+        string="Grouped Repairs",
+        help="All other repairs that belong to the same group.",
+    )
+
+    @api.depends("group_id", "group_id.repair_ids")
+    def _compute_grouped_repair_ids(self):
+        """Compute other repairs in the same group (excluding current repair)."""
+        for repair in self:
+            if not repair.group_id:
+                repair.grouped_repair_ids = False
+                continue
+            repair.grouped_repair_ids = repair.group_id.repair_ids - repair
+
+    def action_add_another_repair(self):
+        """Create a new empty repair linked to the same group.
+
+        If the current repair is not yet assigned to a group, a new
+        `repair.order.group` is automatically created first.
+        The new repair will reuse basic logistics fields such as
+        partner, company, and locations, but not copy any parts or products.
+        """
+        self.ensure_one()
+        if not self.group_id:
+            self.group_id = self.env["repair.order.group"].create({})
+
+        # Create new empty repair record within the same group
+        new_repair = self.env["repair.order"].create(
+            {
+                "group_id": self.group_id.id,
+                "partner_id": self.partner_id.id,
+                "company_id": self.company_id.id,
+                "picking_type_id": self.picking_type_id.id,
+                "location_id": self.location_id.id,
+                "location_dest_id": self.location_dest_id.id,
+            }
+        )
+
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "repair.order",
+            "res_id": new_repair.id,
+            "view_mode": "form",
+            "target": "current",
+        }
+
+    def write(self, vals):
+        """Propagate partner changes across all repairs in the same group.
+
+        When a user changes the partner on one repair, all other repairs
+        of the same group automatically receive the same partner, ensuring
+        data consistency. Recursion is prevented via the context flag
+        `skip_group_sync`.
+        """
+        res = super().write(vals)
+        if "partner_id" in vals and not self.env.context.get("skip_group_sync"):
+            for rec in self:
+                if rec.group_id:
```

Comment:
Why do you need this check if you are checking for rec.grouped_repair_ids two lines below?